### DISCUSSION TITLE
issues with new build and ./configure --with-ssl

### DISCUSSION THREAD

**Discussion by gitchygirl** [curl/curl#12433] [Q&A]
Is this the right place to ask questions about trying to build curl from source?
Trying to deploy curl 7.76.0 since it introduced a new option called fail-with-body we want to use - this is on rhel7 Linux.
Was able to do it on a test server but prod fails (same OS, kernel etc. so something is different but not sure what).

** on a test system that the compile worked on I just see this in the config.log but it continues and I'm able to run make after
conftest.c:9:28: fatal error: ac_nonexistent.h: No such file or directory
 #include <ac_nonexistent.h>
                            ^
compilation terminated.

** review config.log on my non-working prod server
** Syntax error and first fatal error is on line 154 of 667
|                    Syntax error
configure:4892: gcc -E  conftest.c
In file included from /usr/include/bits/posix1_lim.h:160:0,
                 from /usr/include/limits.h:144,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include/limits.h:168,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include/syslimits.h:7,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include/limits.h:34,
                 from conftest.c:10:
/usr/include/bits/local_lim.h:38:26: fatal error: linux/limits.h: No such file or directory
 #include <linux/limits.h>
                          ^
compilation terminated.

** there is a limits.h in /usr/include/ on both servers and they are the same size and datestamp
** same for the one in /usr/include/c++/4.8.2/tr1/limits.h

** check cpp versions and they are the same
cpp --version
cpp (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44.0.3)

Just not sure how to troubleshoot or where to get help other than perhaps #curl via IRC which I'm new to also.

**Comment by dfandrich** [comment]
> Is this the right place to ask questions about trying to build curl from
> source?

The curl-library mailing list is probably better.

> Trying to deploy curl 7.76.0 since it introduced a new option called

Why that version? It's 2.5 years old and contains 35 known security
vulnerabilities. 

> fail-with-body we want to use - this is on rhel7 Linux.

> ** on a test system that the compile worked on I just see this in the
> config.log but it continues and I'm able to run make after
> conftest.c:9:28: fatal error: ac_nonexistent.h: No such file or directory
> #include <ac_nonexistent.h>

This is normal. autoconf is testing that a nonexistent header returns an error.

> ** review config.log on my non-working prod server
> ** Syntax error and first fatal error is on line 154 of 667
> | Syntax error
> configure:4892: gcc -E conftest.c
> In file included from /usr/include/bits/posix1_lim.h:160:0,
> from /usr/include/limits.h:144,
> from /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include/limits.h:168,
> from /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include/syslimits.h:7,
> from /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include/limits.h:34,
> from conftest.c:10:
> /usr/include/bits/local_lim.h:38:26: fatal error: linux/limits.h: No such file
> or directory
> #include <linux/limits.h>
> ^
> compilation terminated.
> 
> ** there is a limits.h in /usr/include/ on both servers and they are the same
> size and datestamp

/usr/include/limits.h isn't the same as /usr/include/linux/limits.  But, this
is just a test to see if a header file is there or not. If not, then configure
can raise an error (if it's required) or work around the missing header (if
it's not).

> Just not sure how to troubleshoot or where to get help other than perhaps #curl
> via IRC which I'm new to also.

So far, I don't see an actual problem that you've raised. The whole point of
configure is to see which header files and features are available on the system
so it can know how to configure the build. Some of these tests will fail (and
you've even discovered one that is *designed* to fail) and that's fine; that's
how it knows if these features are there or not. If the resulting configuration
builds and runs, then it has performed its function as expected.

**Reply by gitchygirl** [reply]
Thanks for the informative reply.  I will provide some answers here:
As far as version goes that was what was requested and I didn't inquire or even think of going with something later and thanks for the feedback on rel.

The actual error I get is this and I guess it seems I left that important piece of information out.
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking how to run the C preprocessor... /lib/cpp
configure: error: in `/root/curl-7.76.0':
configure: error: C preprocessor "/lib/cpp" fails sanity check

So based on the above should I post/search on curl-library instead?

**Reply by bagder** [reply]
> configure: error: C preprocessor "/lib/cpp" fails sanity check

That's the key. configure tries your `cpp` but can't figure out how to use it. Or maybe the installation is incomplete/broken somehow?

**Reply by gitchygirl** [reply]
ok thanks for the feedback.  I did see somewhere else that suggested a reinstall of cpp, I was just a little hesitant to do that on a prod system but might come to that.
